tensorboard/webapp/plugins/npmi/views/annotations_list/annotations_list_toolbar/annotations_list_toolbar_container.ts

```diff
@@ -0,0 +1,107 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, ChangeDetectionStrategy, Input} from '@angular/core';
+import {Store} from '@ngrx/store';
+import {State} from '../../../../../app_state';
+import {map} from 'rxjs/operators';
+import * as npmiActions from '../../../actions';
+
+import {
+  getSelectedAnnotations,
+  getAnnotationsExpanded,
+  getShowCounts,
+  getAnnotationsRegex,
+  getShowHiddenAnnotations,
+} from '../../../store';
+
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
+
+@Component({
+  selector: 'npmi-annotations-list-toolbar',
+  template: `
+    <npmi-annotations-list-toolbar-component
+      [numAnnotations]="numAnnotations"
+      [expanded]="expanded"
+      [selectedAnnotations]="selectedAnnotations$ | async"
+      [annotationsExpanded]="annotationsExpanded$ | async"
+      [showCounts]="showCounts$ | async"
+      [showHidden]="showHidden$ | async"
+      (onFlagAnnotations)="flagAnnotations($event)"
+      (onHideAnnotations)="hideAnnotations($event)"
+      (onToggleExpanded)="toggleExpanded()"
+      (onToggleShowCounts)="toggleShowCounts()"
+      (onToggleShowHidden)="toggleShowHidden()"
+    ></npmi-annotations-list-toolbar-component>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AnnotationsListToolbarContainer {
+  @Input() numAnnotations!: number;
+  @Input() expanded!: boolean;
+  readonly selectedAnnotations$ = this.store.select(getSelectedAnnotations);
+  readonly annotationsExpanded$ = this.store.select(getAnnotationsExpanded);
+  readonly showCounts$ = this.store.select(getShowCounts);
+  readonly showHidden$ = this.store.select(getShowHiddenAnnotations);
+  readonly annotationsFilter$ = this.store.select(getAnnotationsRegex);
+  readonly isAnnotationsFilterValid$ = this.annotationsFilter$.pipe(
+    map((filterString) => {
+      try {
+        // tslint:disable-next-line:no-unused-expression Check for validity of filter.
+        new RegExp(filterString);
+        return true;
+      } catch (err) {
+        return false;
+      }
+    })
+  );
+
+  constructor(private readonly store: Store<State>) {}
+
+  filterChange(filter: string) {
+    this.store.dispatch(
+      npmiActions.npmiAnnotationsRegexChanged({regex: filter})
+    );
+  }
+
+  flagAnnotations(annotations: string[]) {
+    this.store.dispatch(
+      npmiActions.npmiToggleAnnotationFlags({
+        annotations,
+      })
+    );
+    this.store.dispatch(npmiActions.npmiClearSelectedAnnotations());
```

Comment:
This synchronous dispatch of multiple actions are disallowed per our Ngrx style guideline (not written yet; sorry :(). Could you instead dispatch just one action? Let us please talk offline.

Reply:
Changed this to be handled by the reducer.